### PR TITLE
Check return value for getProduct() in getPrice().

### PR DESCRIPTION
**Pull request by pbaylies**
If getProduct() returns a NULL value then this code will generate a PHP warning and potentially crash the page; let's not do that, and check our return values before we try to call methods on them.

### CODE REVIEW

**Review by antonkril — CHANGES_REQUESTED** [review]
Please consider saving results of getCustomOption() and getProduct() calls. Also please address static test feedback.

File: app/code/Magento/ConfigurableProduct/Model/Product/Type/Configurable/Price.php [hunk 1]
```diff
@@ -39,10 +39,15 @@ public function getFinalPrice($qty, $product)
      */
     public function getPrice($product)
     {
-        if ($product->getCustomOption('simple_product')) {
-            return $product->getCustomOption('simple_product')->getProduct()->getPrice();
-        } else {
-            return 0;
+        if ( !empty( $product ) ) {
```
Comment (antonkril) [inline app/code/Magento/ConfigurableProduct/Model/Product/Type/Configurable/Price.php]: This check is not required

**Review by antonkril — APPROVED** [review]

**Comment by vrann** [comment]
@pbaylies Thank you for contribution. Can you please fix following code styles issues:

```
FILE: ...e/Magento/ConfigurableProduct/Model/Product/Type/Configurable/Price.php
--------------------------------------------------------------------------------
FOUND 6 ERROR(S) AFFECTING 3 LINE(S)
--------------------------------------------------------------------------------
 42 | ERROR | Expected 0 spaces after opening bracket; 1 found
 42 | ERROR | Expected 0 spaces before closing bracket; 1 found
 44 | ERROR | Expected 0 spaces after opening bracket; 1 found
 44 | ERROR | Expected 0 spaces before closing bracket; 1 found
 46 | ERROR | Expected 0 spaces after opening bracket; 1 found
 46 | ERROR | Expected 0 spaces before closing bracket; 1 found
--------------------------------------------------------------------------------
```

**Comment by pbaylies** [comment]
@vrann I can't reproduce this; I don't see any extra spaces at the end of those lines on my committed file.

**Comment by vrann** [comment]
@pbaylies 

these lines
```
if ( !empty( $product ) ) {
if ( !empty( $simpleProductOption ) ) {
if ( !empty( $simpleProduct ) ) {
```
should look like this:
```
if (!empty($product)) {
if (!empty($simpleProductOption)) {
if (!empty($simpleProduct)) {
```

**Comment by pbaylies** [comment]
@vrann Done, thanks!

**Comment by vrann** [comment]
@pbaylies Thank you!

**Comment by vrann** [comment]
@pbaylies merged!

**Pull request merged**